include/exadg/fluid_structure_interaction_precice/coupling_interface.h

```diff
@@ -0,0 +1,224 @@
+#pragma once
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/vectorization.h>
+
+#include <deal.II/matrix_free/matrix_free.h>
+
+#include <exadg/matrix_free/integrators.h>
+#include <precice/SolverInterface.hpp>
+
+namespace Adapter
+{
+using namespace dealii;
+
+/**
+ * Enum to handle all implemented data write methods one can use
+ */
+enum class WriteDataType
+{
+  undefined,
+  values_on_dofs,
+  values_on_other_mesh,
+  gradients_on_other_mesh,
+  values_on_quads,
+  normal_gradients_on_quads
```

Comment:
If I understand it correctly, `...on_q_points` would be more precise here. First, I thought of quads vs. triangles.